### DISCUSSION TITLE
Can't setup the syslog input

### DISCUSSION THREAD

**Discussion by itshikanov** [martin-helmich/prometheus-nginxlog-exporter#183] [Q&A]
version: 1.8.0

cat config.hcl 

    listen {
      port = 4040
    }

    namespace "nginx" {
      source = {
        syslog {
          listen_address = "udp://127.0.0.1:8513"
          format = "auto"
        }
      }

    format = "$remote_addr\t$remote_user\t$request\t$status\t$body_bytes_sent\t$http_referer\t$http_user_agent\t$http_x_forwarded_for"
    }

Nginx:
`
log_format syslog_format  '$remote_addr\t$remote_user\t$request\t$status\t$body_bytes_sent\t$http_referer\t$http_user_agent\t$http_x_forwarded_for';
....
access_log syslog:server=127.0.0.1:8513,facility=local7,tag=nginx,severity=info syslog_format;
`

syslog string  from ngrep:
`
U 127.0.0.1:57279 -> 127.0.0.1:8513 #1034
  <190>Feb 18 09:11:00 test-k8s-1 nginx: 185.132.196.98.-.HEAD / HTTP/1.1.200.0.-.curl/7.58.0.-
`

String from strace:
`
[pid  1669] <... recvfrom resumed> "<190>Feb 18 09:11:48 test-k8s-1 nginx: 185.132.196.98\t-\tHEAD / HTTP/1.1\t200\t0\t-\tcurl/7.58.0\t-", 65536, 0, {sa_family=AF_INET, sin_port=htons(57279), sin_addr=inet_addr("127.0.0.1")}, [112->16]) = 93
`

Result:
 HELP nginx_parse_errors_total Total number of log file lines that could not be parsed
TYPE nginx_parse_errors_total counter
nginx_parse_errors_total 0

What wrong?

**Comment by martin-helmich** [comment]
Huh. 🤔  I'm unsure, actually. In general, the syslog listener is verified to be working by automated tests. Are you seeing any output on the exporter's stdout/stderr streams? Have you tried explicitly setting the syslog format (most possibly from `auto` to `rfc3164`)?

**Reply by itshikanov** [reply]
No output in stdout/stderr. 
The thirst time i  was used rfc3164, not working

**Comment by itshikanov** [comment]
```
		for _, f := range slCfg.Tags {
			t, err := tail.NewSyslogFollower(f, server, channel)
			if err != nil {
				panic(err)
			}

			t.OnError(func(err error) {
				panic(err)
			})

			followers = append(followers, t)
		}
```
config
```
namespace "nginx" {
  source = {
    syslog {
      listen_address = "udp://127.0.0.1:8513"
      format = "auto"
    }
  }
```

tags = ["nginx"]   must exist in syslog config.


In documentation, for Kubernetes section,  config-map has no tags line.

**Reply by charlienegri** [reply]
I have the exact same problem (in https://github.com/martin-helmich/prometheus-nginxlog-exporter/issues/184 too) and I have set 
```
kind: ConfigMap
metadata:
  name: nginx-exporter-sidecar-config
data:
  config.hcl: |
    listen {
      port = 4040
    }

    namespace "nginx" {
      source = {
        syslog {
          listen_address = "udp://127.0.0.1:5531"
          format = "auto"
          tags = [
            "nginx"
          ]
        }
      }

      format = "$remote_addr - $remote_user [$time_local] \"$request\" $status $body_bytes_sent \"$http_referer\" \"$http_user_agent\" \"$http_x_forwarded_for\""

      labels {
        app = "hello-world-test"
      }
    }

```
(tried also rfc3164)

**Reply by charlienegri** [reply]
I am trying this in a sidecar on kubernetes and can't figure out what's wrong, I get the same output as above:
```
# HELP nginx_parse_errors_total Total number of log file lines that could not be parsed
# TYPE nginx_parse_errors_total counter
nginx_parse_errors_total 0
```

if I run `kubectl logs mypod -c maincontainer` I see the logs
if I run `kubectl logs mypod -c sidecar` I see
```
loading configuration file /etc/prometheus-nginxlog-exporter/config.hcl
using configuration {Listen:{Port:4040 Address:0.0.0.0 MetricsEndpoint:/metrics} Consul:{Enable:false Address: Datacenter: Scheme: Token: Service:{ID: Name: Address: Tags:[]}} Namespaces:[{Name:nginx NamespaceLabelName: NamespaceLabels:map[] MetricsOverride:<nil> NamespacePrefix: SourceFiles:[] SourceData:{Files:[] Syslog:0xc0000ab040} Parser: Format:$remote_addr - $remote_user [$time_local] "$request" $status $body_bytes_sent "$http_referer" "$http_user_agent" "$http_x_forwarded_for" Labels:map[app:hello-world-test] RelabelConfigs:[] HistogramBuckets:[] PrintLog:false OrderedLabelNames:[] OrderedLabelValues:[]}] EnableExperimentalFeatures:false EnableExperimentalFeaturesOld:false}
starting listener for namespace nginx
running HTTP server on address 0.0.0.0:4040, serving metrics at /metrics
running Syslog server on address udp://127.0.0.1:5531
```

**Reply by MarioCSilva** [reply]
Same thing here, anyone resolved?
Edit: solved after setting the tags on the exporter configmap

**Reply by 2nikitinalexandr2** [reply]
@MarioCSilva would you be so kind as to tell me more precisely 
what tags did I add and where exactly to solved same thing
